lib/webidl2.js

```diff
@@ -407,29 +407,27 @@
     }
 
     function const_value() {
-      const token = consume("true", "false", "null", "Infinity", "-Infinity", "NaN", FLOAT, INT);
-      if (!token) {
-        return;
-      }
-      const { trivia } = token;
-      let data;
+      return consume("true", "false", "null", "Infinity", "-Infinity", "NaN", FLOAT, INT);
```

Comment:
`FLOAT, INT` add indirection, why not just inline them? or do they have some special definition?

Reply:
It's LONG and used twice. Not great to inline IMO.

Reply:
Ok, I figured:) thought I’d ask.

Reply:
Oops, maybe I misunderstood, I thought you asked about inlining `const_value()` which contains them.

For `FLOAT`/`INT`/`STR`/`ID`, they have slightly different meaning. Typically `consume("text")` means consuming a token which is exactly `text`, but in those cases regex is used instead.

Reply:
ah, right. Sorry, I was reviewing on my phone and should have checked that. Using regexes 👍 makes sense then.